src/other/index.jl

```diff
@@ -138,66 +138,66 @@ function Base.insert!(x::Index, idx::Integer, nm::Symbol)
 end
 
 @inline Base.getindex(x::AbstractIndex, idx::Bool) = throw(ArgumentError("invalid index: $idx of type Bool"))
-@inline Base.getindex(x::AbstractIndex, idx::Integer) = Int(idx)
-@inline Base.getindex(x::AbstractIndex, idx::AbstractVector{Int}) = idx
-@inline Base.getindex(x::AbstractIndex, idx::AbstractRange{Int}) = idx
-@inline Base.getindex(x::AbstractIndex, idx::AbstractRange{<:Integer}) = collect(Int, idx)
-@inline Base.getindex(x::AbstractIndex, ::Colon) = Base.OneTo(length(x))
 
-# Fuzzy matching rules:
-# 1. ignore case
-# 2. maximum Levenshtein distance is 2
-# 3. always show matches with 0 difference (wrong case)
-# 4. on top of 3. do not show more than 8 matches in total
-# Returns candidates ordered by (distance, name) pair
-function fuzzymatch(l::Dict{Symbol, Int}, idx::Symbol)
-        idxs = uppercase(string(idx))
-        dist = [(REPL.levenshtein(uppercase(string(x)), idxs), x) for x in keys(l)]
-        sort!(dist)
-        c = [count(x -> x[1] <= i, dist) for i in 0:2]
-        maxd = max(0, searchsortedlast(c, 8) - 1)
-        [s for (d, s) in dist if d <= maxd]
+@inline function Base.getindex(x::AbstractIndex, idx::Integer)
+    if !(1 <= idx <= length(x))
+        throw(BoundsError("attempt to access $(length(x))-element AbstractIndex at index $idx"))
```

Comment:
Maybe this should say "data frame" rather than "AbstractIndex" and "columns" instead of "element"? This is supposed to be seen by end users, right?

Same below.

Reply:
fixed